File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    while (progresses.length > 0) {
        for (let i = 0; i < speeds.length; i++) {
            progresses[i] += speeds[i];
        }

        let count = 0;

        while (progresses.length > 0 && progresses[0] >= 100) {
            progresses.shift();
            speeds.shift();
            count++;
        }

        if (count > 0) {
            answer.push(count);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <queue>
#include <vector>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    queue<int> p;
    queue<int> s;

    for (int x : progresses) p.push(x);
    for (int x : speeds) s.push(x);

    int days = 0;

    while (!p.empty()) {
        int curp = p.front();
        int curs = s.front();
        int count = 0;
        int done = curp + curs * days;
        
        if (done < 100) {
            int remain = 100 - done;
            int addDays = (remain + curs - 1) / curs;
            days += addDays;
        }

        
        while (!p.empty()) {
            int pp = p.front();
            int ss = s.front();

            if (pp + ss * days >= 100) {
                p.pop();
                s.pop();
                count++;
            } else {
                break;
            }
        }

        answer.push_back(count);
    }

    return answer;
}
